### PR TITLE
gh-104886: Remove deprecated configparser.LegacyInterpolation

### LINKED ISSUE #104886: Remove deprecated configparser.LegacyInterpolation class
# Feature or enhancement

The `configparser.LegacyInterpolation` class is undocumented, was deprecated in the docstring since Python 3.2, deprecated with a warning since Python 3.11, and scheduled for removal in 3.13.

# Previous discussion

* https://github.com/python/cpython/issues/90765

### Linked PRs
* gh-104887

### PR DESCRIPTION
**Pull request by hugovk**
Closes #104886.

Remove the undocumented `configparser.LegacyInterpolation` class.

It was deprecated in the docstring since Python 3.2, deprecated with a warning since Python 3.11, and scheduled for removal in 3.13.

* Issue: gh-104886

----
:books: Documentation preview :books:: https://cpython-previews--104887.org.readthedocs.build/

### CODE REVIEW

File: Doc/whatsnew/3.13.rst [hunk 1]
```diff
@@ -118,6 +118,11 @@ Removed
 * Remove support for using :class:`pathlib.Path` objects as context managers.
   This functionality was deprecated and made a no-op in Python 3.9.
 
+* Remove the undocumented :class:`!configparser.LegacyInterpolation` class,
+  deprecated in the docstring since Python 3.2,
+  and with a warning since Python 3.11.
```
Comment (vstinner) [inline Doc/whatsnew/3.13.rst]:
```suggestion
  and with a deprecation warning since Python 3.11.
```

same change in the Changelog entry.

**Review by vstinner — APPROVED** [review]
LGTM, I just had a minor suggestion.

@ambv: Was it you who changed this interpolation method?

Wow, it took a while to completely finish the transition :-)

**Comment by ambv** [comment]
Yeah, I originally never intended to remove the old one but here we are!

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**